### ISSUE TITLE
Restoring open tabs between xed sessions

### ISSUE THREAD

**Issue by Amandaville** [linuxmint/xed#226]
Is there any way to restore open tabs between xed sessions?  I keep 8 to 10 documents open in tabs for reference and updates through the day, and I have to reopen each of them on every restart.  An option or a plugin to do this would be extremely useful.

**Comment by stefanoful** [comment]
HI,
I would also like to keep the files open, even those NOT SAVED ...

**Comment by jpcusumano** [comment]
Agreed. Remembering tabs is the only major thing missing from xed right now, at least given how I use it, which is mostly for sysadmin stuff. It's nice having the most commonly needed system files pop up by default every time I start geany. Something similar under xed would be awesome. I'm using version 1.8.2 under Mint 19.

**Comment by FatBirdie** [comment]
How is this coming along?

**Comment by smokielad** [comment]
I need this so much :-)

**Comment by FredGandt** [comment]
I was also frustrated to find that my opened files were not remembered by xed, and found myself here as I searched for an answer. I look forward to xed having this functionality as standard, but in the meantime, I thought **a not horrible workaround** might prove helpful to others finding their way here as I did.

1. Create a .sh file ("edit.sh" for example) in the directory containing the files you wish to remember as a session. You _can_ put the .sh file somewhere else, but it's simpler this way IMO
2. Open the file in a text editor e.g. xed
3. Add `#!/bin/bash` on the first line
4. Add `xed path/to/file.ext another/path/to/file2.ext flie3.ext` etc. on the next line
5. Save the .sh file
6. Check the file's properties>permissions>allow-executing-file-as-program
7. Double click the .sh file and choose "Run in terminal"

The space separated named files (appropriate encoding required for file names) will be opened in xed, in the tabbed order listed in the command; for example:

`xed index.php styles.css functions.js utils/more-functions.js` will open xed with 4 tabs containing "index.php", "styles.css" and "functions.js" from the directory containing the .sh file just run and "more-functions.js" from the "utils" subdirectory, in that order.

**Comment by Fromax** [comment]
Could this be done with a plugin?

**Comment by dhrn** [comment]
any update on this guys?

Thanks

**Comment by icarter09** [comment]
@dhrn I'm currently focused on getting this functionality implemented

**Comment by MMalakhov** [comment]
So it's been 2 years, and there is no such basic feature like restore the last session. Amazing support!

**Comment by Amandaville** [comment]
I gave up on waiting and moved on to Featherpad when I switched to xfce, and I couldn't be happier.  Featherpad always reopens with the same documents that we're open last time.

**Comment by izmogikan** [comment]
It's a nightmare without "Restoring open tabs" and "navigation between tabs" features.

**Comment by Zardozof** [comment]
I support this request

**Comment by Ben-Fenner** [comment]
Another user wishing for this functionality, be it built-in or via plugin.
I've tried all of the Gedit plugins with no luck:
https://forums.linuxmint.com/viewtopic.php?p=1980782#p1980782

**Comment by FredGandt** [comment]
**Note** the last line focussed in a file is remembered and automatically focussed the next time it's opened, so somewhere, the files' interactions via xed are remembered. Wherever this cache is, there should be a record of the files xed has opened, in order for it to know which line was last focussed in them.
It seems reasonable to suspect that other details such as a timestamp may already be remembered along with this. Considering this state of affairs, it strikes me as possibly being very trivial to implement the requested feature, since the cache already exists for the most part.
If I were more familiar with the xed's code and knew where the cache is, I'd certainly have a stab at implementing this myself.

**Comment by u2n** [comment]
This is the fatal flaw keeping xed from being much more widely adopted. Continuing with Geany for now...

**Comment by FredGandt** [comment]
Switched to the [Atom editor](https://atom.io/). Buh bye xed. R.I.P.

**Comment by HT-7** [comment]
> Is there any way to restore open tabs between xed sessions?

For convenience, I also suggest adding the ability to export or import the session file. Otherwise, I will just get the session file from the configuration directory.

**Comment by tudo75** [comment]
I've ported some of the Gedit plugins to Xed
[https://github.com/tudo75/xed-sessionsaver-plugin](https://github.com/tudo75/xed-sessionsaver-plugin)
[https://github.com/tudo75/xed-terminal-plugin](https://github.com/tudo75/xed-terminal-plugin)
[https://github.com/tudo75/xed-codecomment-plugin](https://github.com/tudo75/xed-codecomment-plugin)

I'm working on to integrate the plugins to make a commit to xed to disribute them with it but it's hard to do

**Comment by Ben-Fenner** [comment]
Great work tudo! I'm trying to install the plugin now and having some issues. I don't have `meson` and I don't have `ninja` so I'm obviously stuck there. I'll make sure I have the required libraries listed and try again.

**Comment by tudo75** [comment]
> Great work tudo! I'm trying to install the plugin now and having some issues. I don't have `meson` and I don't have `ninja` so I'm obviously stuck there. I'll make sure I have the required libraries listed and try again.

Updated dependecies list in readme
Thanks for feedback

**Comment by becker44a** [comment]
I was happy with xed for the most part, except for this missing feature.  As someone migrating from Windows, who used Notepad++ every day, I can say with authority that this feature was important to me.  After 2 1/2 years of apparent inactivity, I have given up and migrated to Geany.  I didn't care for the appearance of any of the available themes (old eyes here), but at least Geany provides means to adjust/customize the colors and snippets.  Bye-Bye xed - such a shame.

**Comment by lowendgamer** [comment]
@tudo75 Please port the restore tab plugin too. I apologize for the inconvenience.

https://github.com/Quixotix/gedit-restore-tabs

**Comment by webarthur** [comment]
I have been using the mousepad text editor, it can save the sessions. After installation, open the editor and go to Edit > Preferences > File > History and set session restore to "always".

**Comment by stonks1334** [comment]
Hi,

Just wondering if this feature was implemented?  Would be great to have tabs restored b/t sessions, especially unsaved tabs.

Thanks.

**Comment by becker44a** [comment]
I don't use xed anymore, but I still watch for that change, and I haven't see any announcement if such a change.

**Comment by tudo75** [comment]
> @tudo75 Please port the restore tab plugin too. I apologize for the inconvenience.
> 
> https://github.com/Quixotix/gedit-restore-tabs

Finally found the time to work on it. You can find it here:
https://github.com/tudo75/xed-restore-tabs-plugin

Hope you like it and feel free to give me all kind of feedback

**Comment by Ben-Fenner** [comment]
@tudo75 I think you've answered the requests of everyone posting in this thread. You've done it! :D

I was able to install the plugin, enable it, and mostly things work great. I do have a couple items of feedback. Would you like those here, or as new issues in the plugin repo?

**Comment by tudo75** [comment]
@Ben-Fenner as issue. I've few time to work on this so better to have things in the right place and don't waste time searching all around

**Comment by m3tabar0n** [comment]
Hi,
is there any progress on this issue?
The plugin from @tudo75 didn't work for me (xed v3.4.3).

**Comment by Ben-Fenner** [comment]
@m3tabar0n I am using `xed` 3.4.3 as well, and Tudo's plugin linked above still works perfectly for me. Are you sure you have it installed properly?

**Comment by m3tabar0n** [comment]
@Ben-Fenner Sorry, I am unable to reproduce atm as I uninstalled the Plugin and moved to **VSCodium** in the meantime.

**Comment by u2n** [comment]
Guess I'm not understanding why the @tudo75 plugin hasn't been incorporated into the main branch and this issue closed.

Can it be that a pull request is needed?

Not being unappreciative here; many thanks to all who contribute. It's just that when the work has been done, why not use it?

**Comment by becker44a** [comment]
Installed and enabled successfully in Xed 3.2.2 on Mint 20.3, Xed 3.4.5 on Mint 21.3, Xed 3.4.3 on LMDE6.  HUGE thanks for this work.  I like to use several different editors, both GUI and CLI, and Xed was dropping off my list due to the lack of this feature.  Using it daily again.  Also agree this work should be merged into Xed for simple access for everyone !

**Comment by VedatDurmaz** [comment]
Many thanks for adding this feature! But is there an essential reason why it couldn't be cherry-picked to version 3.4.3 on Mint 21.2 which based on a much more recent Ubuntu (22.4) version than Mint 20.3 (Ubuntu 20.4)?

**Comment by communist1917** [comment]
Six years left since issue was opened, and developers stil don't give a damn about implementing super easy plugin - remember what files were open at the moment app was closed and reopen them when app is started.

**Comment by dwbapst** [comment]
I just want to throw my hat in to say this feature request is for an extremely valuable attribute, and that @tudo75's port of the gedit plugin to xed works perfectly for me on Mint 22.1 Cinnamon. Thanks Tudo!

**Comment by OlegWebmaster** [comment]
Is it possible to make this a standard plugin in xed? 
_This function is very necessary for xed. I use ArchLinux+Cinnamon._

**Comment by filippologli** [comment]
Thanks so much @tudo75, that's exactly what I'd been searching for. Your deb works fine on LM 22.2.
